pkg/pubsub/adapter/adapter_test.go

```diff
@@ -17,97 +17,355 @@ limitations under the License.
 package adapter
 
 import (
+	"bytes"
 	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	"cloud.google.com/go/pubsub"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/knative-gcp/pkg/apis/events/v1alpha1"
+	"github.com/google/knative-gcp/pkg/pubsub/adapter/converters"
+
 	cloudevents "github.com/cloudevents/sdk-go"
 	cepubsub "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub"
 	pubsubcontext "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub/context"
 )
 
-type mockStatsReporter struct{}
+type mockStatsReporter struct {
+	gotArgs *ReportArgs
+	gotCode int
+}
 
-// TODO: test this more.
 func (r *mockStatsReporter) ReportEventCount(args *ReportArgs, responseCode int) error {
+	r.gotArgs = args
+	r.gotCode = responseCode
 	return nil
 }
 
-// TODO: test this more.
-func TestAdapter(t *testing.T) {
-	ctx := context.TODO()
+func TestStartAdapter(t *testing.T) {
 	a := Adapter{
-		Project:      "proj",
-		Topic:        "top",
-		Subscription: "sub",
+		Project:          "proj",
+		Topic:            "top",
+		Subscription:     "sub",
+		Sink:             "http://localhost:8081",
+		Transformer:      "http://localhost:8080",
+		ExtensionsBase64: "eyJrZXkxIjoidmFsdWUxIiwia2V5MiI6InZhbHVlMiJ9Cg==",
 	}
+	// This test only does sanity checks to see if all fields are
+	// initialized.
+	a.Start(context.Background())
```

Comment:
is start blocking?

Reply:
No. Because it will actually return error as it can't really connect to pubsub.

Reply:
Note added.